### ISSUE TITLE
Improving GraphQL extensions and documentation

### ISSUE THREAD

**Issue by paulpopus** [payloadcms/payload#2557]
# Documentation Issue

I think documentation for extending GraphQL is lacking when it comes to covering a full customisation a user may realistically need. I covered some of this in my tutorial here https://nouance.io/articles/how-to-extend-graphql-in-payload-with-custom-queries-mutations-and-re-usable-types but I want to bring some of those examples and information into the official docs.

Parts that are confusing:
- Available types for my resolvers, return types and so on
- Best practices for code structure
- Explaining how to type and which graphql packages are used and where to dig deeper if you want more power (the source documentation)
- Things to consider when using the LocalAPI within a graphql resolver
  - eg. the issue I struggled with when accidentally setting (or emitting) the `depth` argument 

## Additional Details

Before updating the docs I actually want to expose a few types and utilities from Payload's core.

- [ ] Expose the `buildPaginatedListType` function
- [ ] Expose the type for a GraphQL query or mutation from the internal type `(GraphQL: typeof ImportedGraphQL, payload: Payload) => Record<string, unknown>`
- [ ] Expose Payload's GraphQL type too, currently you have to kind of do this `import type { default as ImportedGraphQL } from 'graphql'`
- [ ] Attach the collection's already built paginatedType to the `.graphQL.` object so that it can be properly re-used in custom queries, making the re-use of fragments in the frontend a lot easier

The goals here are to reduce friction in working with and extending GraphQL. Happy to work on this and add a PR for the code and documentation updates if there's a greenlight from the Payload team

**Comment by paulpopus** [comment]
@ssyberg Yeah, I can see the same issue. Can't figure out why it wouldn't be picked up, maybe someone at payload has a better idea? @denolfe

**Comment by revnelson** [comment]
> Not sure if I'm just missing something, but I had to explicitly import buildPaginated... to get the example working: `import buildPaginatedListType from 'payload/dist/graphql/schema/buildPaginatedListType';`

Thank you for this! The documentation should really be updated until this is fixed.
